### ISSUE TITLE
Add fields to admin exports [13,8,1.6]

### ISSUE THREAD

**Issue by rsgonzal** [Iridescent-CM/technovation-app#2572]
As an admin, I want to be able to pull a more complete export from the platform so that I can have all of the information I need in one place and don't need to combine spreadsheets.

- Want to add additional columns to three pages on TG. I showed Nick three different versions for displaying the information. For Student, Mentor, Parent names & emails on all three pages - he preferred the way it is shown on [submissions 2 ](https://docs.google.com/spreadsheets/d/1mXNgvidI7IuzsbwFNkjEfSJu-REFCpYdpoTihPUJURc/edit#gid=608834310)tab. It's not as clean but easier to use
- Make sure each page lets you select the new fields under 'more columns'

Submission (you can already add student & mentor names but in a different format):
- Student Names (5 cells max)
- Student Emails (5 cells max)
- Mentor Names (one cell separated by commas)
- Mentor Emails (one cell separated by commas)
- Parent Names (5 cells max)
- Parent Emails (5 cells max)
- AI Question
- Climate Question
- Game Question

Team (you can already add student, mentor & parent emails but in a different format):
- Student Names
- Student Emails
- Mentor Names
- Mentor Emails
- Parent Names
- Parent Emails

Scores:
- App Description
- City
- Student Names
- Student Emails
- Mentor Names
- Mentor Emails
- Parent Names
- Parent Emails

## QA STEPS
- As an admin, make sure the 👆 additional fields 👆 are all there and returning the expected results - I think I got them all, let me know if I missed any, it's pretty easy to add more too 😄 
- Also, check out these new fields as a ChA - make sure the ones we want are there and the ones we don't aren't 😄

**Comment by hellafitz** [comment]
@rsgonzal are these proposed additions to a default export or additional options to offer?

**Comment by rsgonzal** [comment]
They're additions to exports that already exist. First is from the Submission section of the app and the second is from the Scores.

**Comment by hellafitz** [comment]
Decisions: 
- These fields should be options you can add on to your export under More Columns
- City = Team's city, associated with the submission

Open Qs:
- What is the row and column format exactly? Currently if we have a list of multiple emails in a single column they're just separated by a comma. If this is going to be a single column with all of the team's names and emails, what does that look like specifically? And how do you want to know who is a mentor, who is a team member, who is a parent? [ @rsgonzal to ask Nick or Amy]

**Comment by nickiredescent** [comment]
Open Qs As:

- [Current Submission Tab exports into a format similar to this](https://docs.google.com/spreadsheets/d/1mXNgvidI7IuzsbwFNkjEfSJu-REFCpYdpoTihPUJURc/edit?usp=sharing) (The order of the columns may not be exact, but the idea is the same)
 - [Current Scores Tab exports into a format similar to this](https://docs.google.com/spreadsheets/d/1mXNgvidI7IuzsbwFNkjEfSJu-REFCpYdpoTihPUJURc/edit#gid=1290008141&range=A1) (The order of the columns may not be exact, but the idea is the same)
 - [Current Teams Tab exports into this format](https://docs.google.com/spreadsheets/d/1mXNgvidI7IuzsbwFNkjEfSJu-REFCpYdpoTihPUJURc/edit#gid=340626318&range=A1)

So for the [Submissions pull](https://docs.google.com/spreadsheets/d/1mXNgvidI7IuzsbwFNkjEfSJu-REFCpYdpoTihPUJURc/edit?usp=sharing), you can see that it already has a column for "Student Names" (column AA) and another for "Mentor Names" (column AB). The way this data shows is names separated but commas - this is fine. The Submissions tab does **not** pull their emails.
The [Teams pull](https://docs.google.com/spreadsheets/d/1mXNgvidI7IuzsbwFNkjEfSJu-REFCpYdpoTihPUJURc/edit#gid=340626318&range=A1) _does_ have a column for student emails (column F) and one for mentor emails (column G) - again, separated by commas. It **does not** pull their names (column A pulls the Team name, not its members).

The formatting that's present in one pull and missing in the other, that's what I want to see as opposed to having to combine the two pulls together manually. Or another way to put it:
 - one column for Student Names, separated by commas. Example Formatting: [First Name1] [Last Name1], [First Name2] [Last Name2], [First Name3] [Last Name3]...
 - one column for Mentor Names, separated by commas
 - one column for Parent Names, separated by commas
 - one column for Student Emails, separated by commas. Example Formatting: [Email1], [Email2], [Email3]...
 - one column for Mentor Emails, separated by commas
 - one column for Parent Emails, separated by commas
 - These options are available on the Teams, Submissions, and Scores tabs.

Did I explain this clearly?

**Comment by hellafitz** [comment]
Thanks, @nickiredescent. Do you want the emails in the same order as the names (to facilitate matching)?

**Comment by nickiredescent** [comment]
That would be the dream. Thanks!

**Comment by hellafitz** [comment]
@rsgonzal to make an example spreadsheet w/ tab for current export, tab for nick's version, and potentially one more for a suggested/improvement version of nick's

**Comment by hellafitz** [comment]
@rsgonzal please update, thanks!

**Comment by hellafitz** [comment]
@rsgonzal bumping this up

**Comment by rsgonzal** [comment]
Want to add additional columns to three pages on TG. I showed Nick three different versions for displaying the information. For Student, Mentor, Parent names & emails on all three pages - he preferred the way it is shown on [submissions 2 ](https://docs.google.com/spreadsheets/d/1mXNgvidI7IuzsbwFNkjEfSJu-REFCpYdpoTihPUJURc/edit#gid=608834310)tab. It's not as clean but easier to use

Submission (you can already add student & mentor names but in a different format):
- Student Names
- Student Emails
- Mentor Names
- Mentor Emails
- Parent Names
- Parent Emails

Team (you can already add student, mentor & parent emails but in a different format):
- Student Names
- Student Emails
- Mentor Names
- Mentor Emails
- Parent Names
- Parent Emails

Scores:
- App Description
- City
- Student Names
- Student Emails
- Mentor Names
- Mentor Emails
- Parent Names
- Parent Emails

**Comment by hellafitz** [comment]
For this format, we need to know how many columns to hold open for each participant type (e.g. max 5 name slots for students) Open Qs: how many mentor columns?

**Comment by nickiredescent** [comment]
> how many mentor slots? We don't (think we) have a limit in place on the platform.

Ah I see the problem if we need to input a strict cap for slots. Since mentors don't have a limit, it could get messy. There were 29 teams last year with more than 5 mentors (one had 11...). **If I had to pin a number, I'd pick 15.** Otherwise, Rox's Solution #1 was a close second choice for me and should alleviate this problem. 


> do all of these things need to be individually selected in Other (e.g. include Student Name 1, Student Name 2, ... 5)? We don't have an existing pattern for selecting 'Student Names' and then adding multiple columns to the export.

I wouldn't mind if these options just give me the whole enchilada of emails
![image](https://user-images.githubusercontent.com/45576078/96918804-b8cfb200-145f-11eb-9691-5accce517aa1.png)


> what specifically is Nick's workflow with the export file (e.g. download as csv, upload csv to Google Sheets)?
Based on the task, select the appropriate columns from the admin panel.
Download as csv, upload to Sheets, manipulate as necessary.
Not super sure how to go in depth with this question. Like, an RA can ask for emails for students and mentors just outside of their region because their Cha Account restricts them to only their state. I'd go through the admin, searching for the accounts in the valid states, and add any columns I might need (email, city, etc.).

**Comment by hellafitz** [comment]
@nickiredescent we could probably put all the mentor names into one cell with a standard character to separate them. In Sheets or Excel you'd need to insert some blank columns then use text-to-columns to extract each name (or email) into its own cell. How does that sound?

**Comment by nickiredescent** [comment]
That sounds good. I'll take that!

**Comment by hellafitz** [comment]
@rsgonzal this seems like it's good to go if we synthesize and pull out the deliverables

**Comment by rsgonzal** [comment]
Add AI Answers to submissions export on screen and download

**Comment by shaunxp20** [comment]
This one is on QA, ready for a looksie! 😄

**Comment by shaunxp20** [comment]
Just realized these new fields will be available to chapter ambassadors as well; I need to make an update to make them only available to admins.

**Comment by shaunxp20** [comment]
I restricted some of these fields to admin only, and others I made available to ChAs (just used my best judgement). We can enable or disable any of them for ChAs pretty easily, just need to know which ones. 👍 😄

**Comment by rsgonzal** [comment]
wooo! thanks @shaunxp20 was there no way to make 'student names' just pop out all of the names?

i think that's ok, it might be nice to have both options.

**Comment by shaunxp20** [comment]
> wooo! thanks @shaunxp20 was there no way to make 'student names' just pop out all of the names?

Hmm, I'm not sure what "pop out all of the names" means, also for which section (submissions, teams, scores)? 🤔 😕 😄

**Comment by rsgonzal** [comment]
For example on submissions, Student emails gives you all the emails in one cell or you can click each one individually. Wondering if you can get them all indivdually but by only clicking one thing.. If that makes any more sense. Kind of what nick says here:

**Comment by shaunxp20** [comment]
> For example on submissions, Student emails gives you all the emails in one cell or you can click each one individually. Wondering if you can get them all individually but by only clicking one thing.. If that makes any more sense. Kind of what nick says here:
> 
> > I wouldn't mind if these options just give me the whole enchilada of emails
> > ![image](https://user-images.githubusercontent.com/45576078/96918804-b8cfb200-145f-11eb-9691-5accce517aa1.png)

Hmm, I might be still confused, like one (selectable) field called "Emails" that will return all student, mentor, and parent email addresses in one column? 🤔 
And for for "Names" that will return all student, mentor, and parent names? 🤔

**Comment by rsgonzal** [comment]
>Hmm, I might be still confused, like one (selectable) field called "Emails" that will return all student, mentor, and parent email addresses in one column? 🤔
And for for "Names" that will return all student, mentor, and parent names? 🤔

no, more like student names would give you the 5 cells, each filled with a different name.. we can talk about it tomorrow!

**Comment by rsgonzal** [comment]
thanks @shaunxp20 ! so from what i gathered from the last meeting with the team, these look great. Can we add the 5 cells for parent emails, parent names, student emails and student names to the **Team** and **Score** sections as well. Just how you have them on the submission page!

**Comment by shaunxp20** [comment]
> Can we add the 5 cells for parent emails, parent names, student emails and student names to the **Team** and **Score** sections as well. Just how you have them on the submission page!

These changes are on QA now, I think I got them all (and they should only be available to admins and not ChAs), let me know if I missed any! 😀

**Comment by rsgonzal** [comment]
@nickiredescent @ophelieh can you two take a look at this one please?

Ophelie, can you take a look at the admin and then see if you want any of those extra fields for ChAs please?

**Comment by nickiredescent** [comment]
**Submissions Tab**
The Additional Information Q's fields are working as expected.
Student names and emails look good
Parents names and emails look good
Mentor names are appearing as the same as their email. I can confirm that all these mentors should have names that are unique from their email (for example, mackenzie+8@technovation.org's name should be "Margaret Dancho").
![image](https://user-images.githubusercontent.com/45576078/113231278-bca60280-924f-11eb-838b-86d1f6a7abcc.png)

**Teams Tab**
All names and email fields look good (There is no issue with mentor names here).

**Scores Tab**
All names and email fields look good (There is no issue with mentor names here).

**Comment by shaunxp20** [comment]
> Mentor names are appearing as the same as their email.

Thank you for looking at this, Nick! And good catch! This should be fixed on QA now! 👍 😁

**Comment by rsgonzal** [comment]
Thanks for fixing, shaun! looks good. If there is any addition fields we want for ChAs, we can open another ticket

**Comment by ophelieh** [comment]
Sorry I ran out of time but will look at it by Monday latest!

**Comment by ophelieh** [comment]
Hi! 

I am looking at the submission tab right now and I have a feeling this could get messy (same for the others with the same format). I skimmed through all the comments, couldn't find the reason we separated each student name, and each student email? I feel like it could lead to more human error than having all the emails and names pulled at once - for example not pulling the name for student 2 but pulling the email. I saw that there is also the previous option to have all students names and emails in a column which is great. I'm just not sure why we are adding the extra option? 

Reagarding chapter ambassadors, the only options in each section (submission, score, team) I would offer is the names and emails divided by account type not Student 1 email... I don't think it would be that helpful to them. 

On the scores' tab, love being able to have the student names and emails right there!!! :) It will be so helpful! 

Is there a way to cut off part of the app description and add [...] or anything else to avoid the huge cells? 
![Screen Shot 2021-04-05 at 4 51 11 PM (2)](https://user-images.githubusercontent.com/32206421/113631473-24cd5d80-962f-11eb-91e8-5dc1c82a3ad2.png)

Thanks!
